libappstream-glib/as-utils.c

```diff
@@ -2015,3 +2015,38 @@ as_utils_appstream_id_valid (const gchar *str)
 	}
 	return TRUE;
 }
+
+/**
+ * as_utils_version_to_double:
+ * @str: a string
+ *
+ * Returns an approximation to a version string that can be expressed as a double.
+ * The approximation is equal or less than the given version string.
+ *
+ * Returns: the approximation.
+ *
+ * Since: 0.6.4
+ */
+gdouble
+as_utils_version_to_double (const gchar *version)
+{
+	const gchar *minor;
+
+	minor = strchr (version, '.');
+	if (minor == NULL)
+		return g_ascii_strtod (version, NULL);
+	minor += 1;
+
+	// A floating point number can only correctly represent a
+	// minor version that is a single digit.  Anything else is
+	// approximated by 0.9.  Anything after the minor version is
+	// ignored.
+	//
+	// For example, the string "0.7.5" is returned as 0.7, while
+	// "0.11" is returned as 0.9.
+
```

Comment:
Rather than doing this, can't we just mandate that appstream versions have three components, e.g. 0.6.0, 0.11.4 or 1.0.0?

Reply:
Can you elaborate a bit?  What do refer to with "doing this"?  What does mandating three components change?  Does it simplify the code in `as_utils_version_to_double`?

Reply:
Ahh, it's only used for the now-deprecated function. Gotcha.

Reply:
The libappstream library uses an enum for this, so I don't have to parse a string or do weird version comparisons. Maybe worth considering here as well.
See https://github.com/ximion/appstream/blob/master/src/as-metadata.h#L90

Reply:
An enum (that enum?) works for me.

Reply:
I have no opinion and I am happy to change the code to use an enum. @hughsie, can you make the call?